### PR TITLE
주문 조회 및 주문 메뉴 CRUD 도메인 로직 구현

### PR DESCRIPTION
**Pull request by siwan9**
1️⃣ 영수증 별 주문 목록 조회(대상 영수증의 고객과 점주 권한 필요)
--- 
image

2️⃣ 주문 상세 조회(대상 영수증의 고객과 점주 권한 필요)
---
image

3️⃣ 주문 메뉴 추가(대상 영수증의 점주 권한 필요)
---
image

4️⃣ 주문 메뉴 수량 수정(대상 영수증의 점주 권한 필요)
---
image

5️⃣ 주문 메뉴 삭제(대상 영수증의 점주 권한 필요)
---
image

✅ 주문 메뉴 API는 예기치 못한 상황에 점주가 주문을 컨트롤 할 수 있도록 하기 위해 만든 API 이므로, 점주만 가능하도록 권한을 설정하였습니다.

## Summary by CodeRabbit

- **New Features**
  - Added support for managing order menu items, including creation, quantity updates, and deletion.
  - Introduced new error messages for user access denial, closed sales, and missing order menu items.
  - Implemented customer registration for receipts and expanded order retrieval options by receipt or order ID.
  - Added user passport validation to ensure user identity consistency.

- **Enhancements**
  - Improved validation for sale status and ownership when performing order and receipt operations.
  - Refined error handling for order, menu, and receipt operations with clearer feedback.
  - Delegated ownership and role validation to dedicated components for receipts and stores.
  - Updated receipt and sale retrieval methods to more granular variants reflecting store and sale status.

- **Bug Fixes**
  - Addressed edge cases for access control and entity existence during order and receipt processing.

- **Tests**
  - Added comprehensive tests for order menu and order services, including new scenarios and validation logic.
  - Updated receipt service tests to cover enhanced validation and ownership checks.
  - Removed outdated receipt validator tests.

- **Chores**
  - Refactored and streamlined entity constructors, factory methods, and test fixtures for better maintainability.
  - Removed user passport references from receipt entity and fixtures.

### CODE REVIEW

File: domain/domain-pos/src/main/java/domain/pos/order/service/OrderService.java [hunk 1]
```diff
@@ -53,10 +52,17 @@ public Order patchOrderStatus(Long orderId, UserPassport userPassport, OrderStat
...
+		Receipt receipt = receiptReader.getReceiptWithCustomerAndOwner(receiptId)
+			.orElseThrow(() -> new ServiceException(ErrorCode.RECEIPT_NOT_FOUND));
+		receiptValidator.validateAccessToReceipt(receipt, userPassport);
+		return orderReader.getReceiptOrders(receiptId);
+	}
+
+	public Order getOrder(Long orderId, UserPassport userPassport) {
+		Order order = orderReader.getOrderWithCustomerAndOwner(orderId)
+			.orElseThrow(() -> new ServiceException(ErrorCode.ORDER_NOT_FOUND));
+		receiptValidator.validateAccessToReceipt(order.getReceipt(), userPassport);
```
Comment (kang20) [inline domain/domain-pos/src/main/java/domain/pos/order/service/OrderService.java:65]: P3: 해당 validate 가 영수증이 비회원 결제이거나 회원결제면서 요청 유저가 소유자 또는 점주일때인데

비회원 결제를 했는데, 요청 유저가 소유자가 아닌 경우 validate 통과되지 않나요

File: domain/domain-pos/src/test/java/domain/pos/order/service/OrderMenuServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+			// then
+			verify(orderReader).getOrderWithCustomerAndOwner(orderId);
+			verify(receiptValidator).validateIsOwner(any(), eq(userPassport));
+			verify(menuReader).getMenuInfo(any(), any());
+			verify(orderMenuWriter).postOrderMenu(orderId, orderMenu);
+		}
+
+		@Test
+		void 주문_조회_실패() {
```
Comment (kang20) [inline domain/domain-pos/src/test/java/domain/pos/order/service/OrderMenuServiceTest.java:73]: p3: 실패 케이스에 메서드 네이밍을 왜 실패했는지 명확히 적으면 좋을거 같아요

File: domain/domain-pos/src/main/java/domain/pos/order/service/OrderMenuService.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		return orderMenuWriter.postOrderMenu(orderId, orderMenu);
+	}
+
+	@Transactional
+	public OrderMenu patchOrderMenuQuantity(Long orderMenuId, UserPassport userPassport, Integer quantity) {
+		OrderMenu orderMenu = orderMenuReader.getOrderMenuWithCustomerAndOwner(orderMenuId)
+			.orElseThrow(() -> new ServiceException(ErrorCode.ORDER_MENU_NOT_FOUND));
+		receiptValidator.validateIsOwner(orderMenu.getOrder().getReceipt(), userPassport);
+		return orderMenuWriter.patchOrderMenuQuantity(orderMenu, quantity);
+	}
```
Comment (kang20) [inline domain/domain-pos/src/main/java/domain/pos/order/service/OrderMenuService.java:47]: P3: 저희 메뉴에도 수량이 존재하나요?

**Comment by siwan9** [comment]
🛠️ 수정 사항
---
image
- 기존 구조 : Receipt 테이블내에 customerId가 컬럼으로 직접 존재하여 해당 customer와 점주만 영수증에서 주문을 넣는 등 상호작용이 가능했음
- 변경 구조
  - 모든 유저가 주문 및 상호작용이 가능하도록 권한을 열었음(사유 : 영수증을 생성한 한 유저만 주문이 가능하도록 하는 것은 사용자 입장에서 많이 불편할 것으로 생각이 되었기 때문)
    - 무단 주문 공격에 대응하기 위해 추후 위치기반으로 가게와 어느정도 가까이 있어야만 주문이 가능하도록 리팩토링 할 예정
  - Receipt 테이블과 연결된 Receipt_Customer 테이블을 생성하여 로그인 한 유저가 동일한 영수증에서 주문할 경우 Receipt_Customer 테이블에 등록되어 추후 영수증 내역 확인 등이 가능하도록 하였음

🍀 서비스 로직 리팩토링
--- 
1️⃣ (OrderService)주문 생성
---
image
- 주문 가능 유저인지 검증하는 로직을 제외
- (1)정산이 안된 영수증을 가져오도록 조건 추가(정산이 완료된 영수증에 주문추가 금지 목적)
- (2)가게영업 여부 검증 로직 및 (4)고객의 조건부 영수증 매핑 로직 추가

2️⃣ (OrderService)주문 상태 수정, 영수증 별 주문 목록 조회, 주문 상세 조회
---
- 요청 가능 유저인지 검증하는 로직을 제외 

3️⃣ (ReceiptService)영수증 상세조회
---
- 요청 가능 유저인지 검증하는 로직을 제외 

4️⃣ (ReceiptService)영업별 영수증 목록조회, 영수증 정산, 영수증 삭제
---
- 요청 유저가 점주인지 검증하는 로직으로 변경 

5️⃣ (SaleRepository)불필요 메서드 삭제
---
- @kang20 중복되는 불필요한 메서드 삭제 완료했습니다

File: domain/domain-pos/src/main/java/domain/pos/order/service/OrderMenuService.java (hunk 3 above)
Comment (siwan9) [reply domain/domain-pos/src/main/java/domain/pos/order/service/OrderMenuService.java]: 주문 메뉴에 수량이 존재합니다!

File: domain/domain-pos/src/test/java/domain/pos/order/service/OrderMenuServiceTest.java (hunk 2 above)
Comment (siwan9) [reply domain/domain-pos/src/test/java/domain/pos/order/service/OrderMenuServiceTest.java:73]: 좀 더 신경써서 네이밍해보도록 하겠습니다

**Review by kang20 — APPROVED** [review]
LGTM!
